diff --git a/src/expression_grammar.cpp b/src/expression_grammar.cpp
index 09234a0..309205b 100644
--- a/src/expression_grammar.cpp
+++ b/src/expression_grammar.cpp
@@ -24,5 +24,15 @@
 // once by src/expression.cpp and once by mapnik/transform_expression_grammar_impl.hpp
 #include <mapnik/expression_grammar_impl.hpp>
 #include <string>
+#include <stdexcept>
 
 template struct mapnik::expression_grammar<std::string::const_iterator>;
+
+// Prevent linking to libstdc++'s __throw_out_of_range_fmt
+// https://github.com/springmeyer/glibcxx-symbol-versioning/blob/master/test_out_of_range.cpp
+void  __throw_out_of_range_fmt(const char*, ...) __attribute__((__noreturn__));
+void  __throw_out_of_range_fmt(const char* err, ...)
+{
+    // Safe and over-simplified version. Ignore the format and print it as-is.
+    std::__throw_out_of_range(err);
+}
\ No newline at end of file
diff --git a/utils/pgsql2sqlite/build.py b/utils/pgsql2sqlite/build.py
index 45d9003..9d909df 100644
--- a/utils/pgsql2sqlite/build.py
+++ b/utils/pgsql2sqlite/build.py
@@ -39,6 +39,20 @@ source = Split(
 program_env['CXXFLAGS'] = copy(env['LIBMAPNIK_CXXFLAGS'])
 program_env['LINKFLAGS'] = copy(env['LIBMAPNIK_LINKFLAGS'])
 program_env.Append(CPPDEFINES = env['LIBMAPNIK_DEFINES'])
+program_env['LIBS'] = []
+
+if env['RUNTIME_LINK'] == 'static':
+    # pkg-config is more reliable than pg_config across platforms
+    cmd = 'pkg-config libpq --libs --static'
+    try:
+        program_env.ParseConfig(cmd)
+    except OSError, e:
+        program_env.Append(LIBS='pq')
+else:
+    program_env.Append(LIBS='pq')
+
+# Link Library to Dependencies
+libraries = copy(program_env['LIBS'])
 
 if env['HAS_CAIRO']:
     program_env.PrependUnique(CPPPATH=env['CAIRO_CPPPATHS'])
@@ -46,9 +60,8 @@ if env['HAS_CAIRO']:
 
 program_env.PrependUnique(CPPPATH=['#plugins/input/postgis'])
 
-libraries = []
 boost_program_options = 'boost_program_options%s' % env['BOOST_APPEND']
-libraries.extend([boost_program_options,'sqlite3','pq',env['MAPNIK_NAME'],'icuuc'])
+libraries.extend([boost_program_options,'sqlite3',env['MAPNIK_NAME'],'icuuc'])
 
 if env.get('BOOST_LIB_VERSION_FROM_HEADER'):
     boost_version_from_header = int(env['BOOST_LIB_VERSION_FROM_HEADER'].split('_')[1])
@@ -59,13 +72,8 @@ if env.get('BOOST_LIB_VERSION_FROM_HEADER'):
 if env['SQLITE_LINKFLAGS']:
     program_env.Append(LINKFLAGS=env['SQLITE_LINKFLAGS'])
 
-if env['RUNTIME_LINK'] == 'static':
-    if env['PLATFORM'] == 'Darwin':
-        libraries.extend(['ldap', 'pam', 'ssl', 'crypto', 'krb5'])
-    else:
-        # TODO - parse back into libraries variable
-        program_env.ParseConfig('pg_config --libs')
-        libraries.append('dl')
+if env['RUNTIME_LINK'] == 'static' and env['PLATFORM'] == 'Linux':
+    libraries.append('dl')
 
 pgsql2sqlite = program_env.Program('pgsql2sqlite', source, LIBS=libraries)
 Depends(pgsql2sqlite, env.subst('../../src/%s' % env['MAPNIK_LIB_NAME']))
