diff --git a/src/algorithm/connection.cpp b/src/algorithm/connection.cpp
index 0d3906f..f76facf 100644
--- a/src/algorithm/connection.cpp
+++ b/src/algorithm/connection.cpp
@@ -31,6 +31,9 @@
 namespace SFCGAL {
 namespace algorithm {
 
+static const size_t INVALID_INDEX = size_t( -1 ) ; // would use std::numeric_limits< size_t >::max() if it were constant, or SIZE_MAX if it were easier to find.
+
+
 void SurfaceGraph::addRing( const LineString& ring, FaceIndex faceIndex )
 {
     const size_t numSegments = ring.numSegments() ;
@@ -151,7 +154,7 @@ bool isClosed( const SurfaceGraph& graph )
     const SurfaceGraph::EdgeMap::const_iterator end = graph.edgeMap().end() ;
 
     for ( SurfaceGraph::EdgeMap::const_iterator e= graph.edgeMap().begin(); e != end; ++e ) {
-        if ( e->second.second == SurfaceGraph::INVALID_INDEX ) {
+        if ( e->second.second == INVALID_INDEX ) {
             return false;
         }
     }
diff --git a/src/algorithm/connection.h b/src/algorithm/connection.h
index ed9627c..7e2a1ff 100644
--- a/src/algorithm/connection.h
+++ b/src/algorithm/connection.h
@@ -44,7 +44,6 @@ public:
     typedef size_t VertexIndex;
     typedef size_t FaceIndex;
     typedef std::map< Coordinate, VertexIndex >  CoordinateMap ;
-    static const size_t INVALID_INDEX = size_t( -1 ) ; // would use std::numeric_limits< size_t >::max() if it were constant, or SIZE_MAX if it were easier to find.
     // an edge is inserted with vtx ordered by the first polygon we treat,
     // we search the edge with reverse ordered vtx indexes.
     // as a result, an inconsistent orientation between polygons can be spotted by
