diff --git a/Makefile b/Makefile
index 523a3d8..d1e7fce 100644
--- a/Makefile
+++ b/Makefile
@@ -57,7 +57,7 @@ V ?= 1
 ifeq ($(BUILDTYPE),Release)
 all: out/Makefile $(NODE_EXE)
 else
-all: out/Makefile $(NODE_EXE) $(NODE_G_EXE)
+all: out/Makefile $(NODE_G_EXE)
 endif
 
 # The .PHONY is needed to ensure that we recursively use the out/Makefile
@@ -733,7 +733,7 @@ endif
 
 $(BINARYTAR): release-only
 	$(RM) -r $(BINARYNAME)
-	$(RM) -r out/deps out/Release
+	$(RM) -r out/deps out/Debug
 	$(PYTHON) ./configure \
 		--prefix=/ \
 		--dest-cpu=$(DESTCPU) \
@@ -902,4 +902,3 @@ endif
         bench-ci lint-js-ci doc-only $(TARBALL)-headers test-ci test-ci-native \
         test-ci-js build-ci test-hash-seed clear-stalled  test-addons-napi \
         build-addons-napi
-
diff --git a/tools/install.py b/tools/install.py
index bb7d528..6dea6d6 100755
--- a/tools/install.py
+++ b/tools/install.py
@@ -110,7 +110,7 @@ def subdir_files(path, dest, action):
 def files(action):
   is_windows = sys.platform == 'win32'
   output_file = 'node'
-  output_prefix = 'out/Release/'
+  output_prefix = 'out/Debug/'
 
   if 'false' == variables.get('node_shared'):
     if is_windows:
@@ -131,7 +131,7 @@ def files(action):
     action([output_prefix + output_file], 'lib/' + output_file)
 
   if 'true' == variables.get('node_use_dtrace'):
-    action(['out/Release/node.d'], 'lib/dtrace/node.d')
+    action(['out/Debug/node.d'], 'lib/dtrace/node.d')
 
   # behave similarly for systemtap
   action(['src/node.stp'], 'share/systemtap/tapset/')
@@ -163,7 +163,7 @@ def headers(action):
 
   # Add the expfile that is created on AIX
   if sys.platform.startswith('aix'):
-    action(['out/Release/node.exp'], 'include/node/')
+    action(['out/Debug/node.exp'], 'include/node/')
 
   subdir_files('deps/v8/include', 'include/node/', action)
 
