diff --git a/configure.ac b/configure.ac
index a633803c..a36a75e8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -50,8 +50,8 @@ AC_SUBST(RAPIDJSON_CPPFLAGS, "$RAPIDJSON_CPPFLAGS -DRAPIDJSON_HAS_STDSTRING")
 
 # Note - because libz4 changed it's versioning scheme, we need to check for both >131, and if
 #        that fails, try >= 1.7.3
-PKG_CHECK_MODULES([DEPS], [protobuf >= 2.4.0 libcurl >= 7.35.0 zlib >= 1.2.8 liblz4 >= 131],, [
-  PKG_CHECK_MODULES([DEPS], [protobuf >= 2.4.0 libcurl >= 7.35.0 zlib >= 1.2.8 liblz4 >= 1.7.3])
+PKG_CHECK_MODULES([DEPS], [protobuf >= 2.4.0 libcurl >= 7.35.0 zlib >= 1.2.3 liblz4 >= 131],, [
+  PKG_CHECK_MODULES([DEPS], [protobuf >= 2.4.0 libcurl >= 7.35.0 zlib >= 1.2.3 liblz4 >= 1.7.3])
 ])
 
 # if we wanted services
